File: irrgartenjaava/src/irrgartenjaava/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjaava;

/**
 *
 * @author jxlig0d
 */
public class Monster extends LabyrinthCharacter {
    
    // El resto de variables las eliminamos porque ya existen en super
    private static final int INITIAL_HEALTH=5;
    
    
    Monster(String name, float intelligence, float strength){
        super(name, intelligence, strength, INITIAL_HEALTH);

    }
    
    @Override
    public float attack(){
        return Dice.intensity(this.getStrength());
    }
    
    @Override
    public boolean defend(float receivedAttack){
        
        boolean isDead=this.dead();
        if(!isDead){
            float defensiveEnergy=Dice.intensity(this.getIntelligence());
            
            if (defensiveEnergy < receivedAttack){
                this.gotWounded();
                isDead=this.dead();
            }
        }
        
        return isDead;
    } 
    
    // Creo que tambien sobra porque en la clase labyrinthcharacter, ya tenemos un tostring, aunque lo dejo aqui por si acaso
    //@Override
    //public String toString(){
    //   return "M[ nombre:" + name + " , inteligencia:" + intelligence + " , fuerza: " + strength + " , salud: " + health + " , row: " + row + " , col: " + col + " ]";
    //}
    
}

File: irrgarten_ruby/src/monster.rb
#encoding:utf-8

require_relative 'dice'
require_relative 'labyrinth_character'

module Irrgarten

  # clase de los monstruos del juego
  class Monster < LabyrinthCharacter

    # salud inicial de los monstruos

    @@INITIAL_HEALTH = 50000.0 # Solo por testeo de fuzzy
    @@INITIAL_STRENGTH = 10000   # Solo por testeo de fuzzy

    ## estos argumentos de row y col no deberia de ponerlos, estoy debugeando
    def initialize(name, intelligence, strength) # Solo por testeo de fuzzy
      super(name, intelligence, @@INITIAL_STRENGTH, @@INITIAL_HEALTH) # SOLO HABIRA QUE QUITAR EL INITIAL STRENGTH
    end                                                                # y pasarle el del parametro
    

    def attack()
      return Dice.intensity(@strength)
    end

    # p3
    def defend(received_attack)

      is_dead=dead

      if (!is_dead)

        defensive_energy=Dice.intensity(@intelligence)
        if (defensive_energy<received_attack)
          got_wounded
          is_dead=dead
        end
      end

      return is_dead
    end
    
    def to_s()
      return "M" + super
    end


  end # class
end # module